lattices/src/map_union.rs

```diff
@@ -131,15 +133,60 @@ where
             }
         }
         match (self_any_greater, other_any_greater) {
-            (true, false) => Some(Ordering::Greater),
-            (false, true) => Some(Ordering::Less),
-            (false, false) => Some(Ordering::Equal),
+            (true, false) => Some(Greater),
+            (false, true) => Some(Less),
+            (false, false) => Some(Equal),
             // We check this one after each loop iteration.
             (true, true) => unreachable!(),
         }
     }
 }
 
+impl<MapSelf, MapOther, K, ValSelf, ValOther> PartialEq<MapUnion<MapOther>> for MapUnion<MapSelf>
+where
+    MapSelf: Map<K, ValSelf, Key = K, Item = ValSelf> + MapIter + SimpleKeyedRef,
+    MapOther: Map<K, ValOther, Key = K, Item = ValOther> + MapIter + SimpleKeyedRef,
+    ValSelf: PartialEq<ValOther>,
+{
+    fn eq(&self, other: &MapUnion<MapOther>) -> bool {
+        if self.0.len() != other.0.len() {
+            return false;
+        }
+
+        for k in self
+            .0
+            .iter()
+            .map(|(k, _v)| <MapSelf as SimpleKeyedRef>::into_ref(k))
+            .chain(
+                other
+                    .0
+                    .iter()
+                    .map(|(k, _v)| <MapOther as SimpleKeyedRef>::into_ref(k)),
+            )
+        {
+            match (self.0.get(k), other.0.get(k)) {
+                (Some(self_value), Some(other_value)) => {
+                    if *self_value != *other_value {
+                        return false;
+                    }
+                }
+                (None, None) => unreachable!(),
+                _ => {
+                    return false;
+                }
+            }
+        }
+
+        true
+    }
+}
+impl<MapSelf, K, ValSelf> Eq for MapUnion<MapSelf>
+where
+    MapSelf: Map<K, ValSelf, Key = K, Item = ValSelf> + MapIter + SimpleKeyedRef,
+    ValSelf: PartialEq,
+{
+}
```

Comment:
`ValSelf: Eq`

could do `Self: PartialEq<Self>` for the complicated bounds I think

Reply:
yep, I think it works.